server/src/main/java/org/apache/iotdb/db/sync/sender/manager/LocalSyncManager.java

```diff
@@ -46,8 +54,31 @@ public LocalSyncManager(DataRegion dataRegion, Pipe pipe) {
 
   /** tsfile */
   @Override
-  public void syncRealTimeDeletion(Deletion deletion) {
-    syncPipe.collectRealTimeDeletion(deletion, dataRegion.getStorageGroupName(), dataRegionId);
+  public void syncRealTimeDeletion(
+      Deletion deletion,
+      List<TsFileResource> unsealedResources,
+      List<TsFileResource> sealedResources) {
+    Set<String> devices = new HashSet<>();
+    unsealedResources.forEach(o -> devices.addAll(o.getDevices()));
+    sealedResources.forEach(o -> devices.addAll(o.getDevices()));
+
+    for (String device : devices) {
+      try {
+        PartialPath path =
+            new PartialPath(device).concatNode(IoTDBConstant.MULTI_LEVEL_PATH_WILDCARD);
+        if (deletion.getPath().overlapWith(path)) {
+          Deletion splitDeletion =
+              new Deletion(deletion.getPath(), 0, deletion.getStartTime(), deletion.getEndTime());
+          syncPipe.collectRealTimeDeletion(
+              splitDeletion, dataRegion.getStorageGroupName(), dataRegionId);
+        }
+      } catch (IllegalPathException e) {
+        logger.error(
+            String.format(
+                "Parse device path %s to partial path error when splitting deletion.", device),
+            e);
```

Comment:
These codes may not  really be understandable, 

If the codes in 69 is true for `device1`, then the deletion will be added to the sync pipe.

In the next loop, for `device2`, if the codes in 69 is still `TRUE`, then the deletion will be added to the sync pipe **again**. is this expected?

Reply:
> These codes may not really be understandable,
> 
> If the codes in 69 is true for `device1`, then the deletion will be added to the sync pipe.
> 
> In the next loop, for `device2`, if the codes in 69 is still `TRUE`, then the deletion will be added to the sync pipe **again**. is this expected?

Yes.. The implementation is incorrect and still WIP. We should put the intersection of `device+".*"` and `deletion.getPath()` to splitDeletion.（#7823）